packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml

```diff
@@ -222,6 +265,16 @@ Resources:
             Condition:
               StringLike:
                 "aws:sourceArn": { "Fn::Sub": "arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:*" }
+      EncryptionConfiguration:
```

Comment:
I think this is a breaking change. Change to `EncryptionConfiguration` requires replacement.

Reply:
Yes it will replace the repository when you decide to change encryption for the repository. Added a note to the cli command.

I would not say that this is a breaking change but a feature that comes at a cost of a new repository.

Reply:
> I would not say that this is a breaking change but a feature that comes at a cost of a new repository.

That is a breaking change, and not something that we can accept. Also the way this has been implemented it is not something that you can _choose_. If you update to a version of the CLI that has this change it will be applied, and there is no way to opt out.

Also, for ECR repositories you cannot delete the repository if there are images in the repository. And even if there was a way that would mean that we are deleting images for production applications.

Reply:
I see what you mean. When we use `AWS::NoValue` we will not force a replacement of the repository. This would make it opt in.

The issue with having to delete images from the repository still exists though. I would say that the user should then think about migration if they explicitly turn on this value.

Another issue would arise with the RepositoryName since it does not support replacement when enabling encryption. Did you guys already handle something like this in the past?

Reply:
This should work. See my new comment on the defaults. I think we should have 3 values here.